xayn-ai/src/reranker/sync.rs

```diff
@@ -0,0 +1,27 @@
+use crate::{data::UserInterests, error::Error};
+use serde::{Deserialize, Serialize};
+
+/// Synchronizable data of the reranker.
+#[cfg_attr(test, derive(Clone, PartialEq, Debug))]
+#[derive(Default, Serialize, Deserialize)]
+pub(crate) struct SyncData {
+    pub(crate) user_interests: UserInterests,
+}
+
+impl SyncData {
+    pub(crate) fn deserialize(bytes: &[u8]) -> Result<Self, Error> {
+        if bytes.is_empty() {
+            return Ok(Self::default());
```

Comment:
With this we risk to override valid data. Is there an use case to deserialize and empty slice?

Reply:
I believe there is not, so it's removed now.